source/dagonBackend.d

```diff
@@ -1381,6 +1385,8 @@ final class SacScene: Scene{
 			if(pressed(options.hotkeys.moveBackward)) dir += forward;
 			if(pressed(options.hotkeys.turnLeft)) dir += -right;
 			if(pressed(options.hotkeys.turnRight)) dir += right;
+			if(pressed(options.hotkeys.moveUp)) dir += -upVector;
+			if(pressed(options.hotkeys.moveDown)) dir += upVector;
```

Comment:
This looks a bit weird. It seems this mixes up `up` and `down`?

Reply:
i had it the other way around initially and then it worked wrong.

do the coordinates rise as you go towards the sky, or do they fall?
as in "is the left-topmost point of the enveloping rectangle that contains equal to (0,0) or some other coordinates?"

if it's (0,0) and the coords decrease as you move up, then it makes sense. otherwise - i got no good explanation why it works. other than maybe mixing up keybindings, which seem to have a different problem too.

originally i had them set up as 
```
Camera Move Up: Ctrl-q
Camera Move Down: Ctrl-e
```

but they worked fine with just 'q' and 'e' being pressed without the ctrl.

Reply:
`Z` increases towards the sky. Maybe rename `upVector` to `down` then.

Movement hotkeys currently ignore modifiers, because that is what original sac also does for the standard movement hotkeys.

Reply:
```
			// Normal (camera-dependant) movement
			if(pressed(options.hotkeys.moveForward)) dir += -forward;
			if(pressed(options.hotkeys.moveBackward)) dir += forward;
			if(pressed(options.hotkeys.turnLeft)) dir += -right;
			if(pressed(options.hotkeys.turnRight)) dir += right;
			if(pressed(options.hotkeys.moveUp)) dir += -upVector;
			if(pressed(options.hotkeys.moveDown)) dir += upVector;
```

i don't think it's mixed up. seems like moving forward/right/up is the negative side of things.

Reply:
I mean, do you think it'd look better like this?

```
		Vector3f forward = fpview.camera.worldTrans.forward;
		Vector3f right = fpview.camera.worldTrans.right;
		Vector3f downVector = fpview.camera.worldTrans.up;
		Vector3f cameraForward = fpview.camera.observerTrans.forward;
		Vector3f cameraRight = fpview.camera.observerTrans.right;
		Vector3f cameraDownVector = fpview.camera.observerTrans.up;
		. . .
			if(pressed(options.hotkeys.moveUp)) dir += -downVector;
			if(pressed(options.hotkeys.moveDown)) dir += downVector;
```

Reply:
Yes, but it would be better to call it just `down`, not `downVector`. I guess we can fix the fact that dagon calls it `up` some other time.

Reply:
oh you mean shorten it. alright, that i could do. it just looks weird to have something like this:
```
down = foo.bar.blabla.up;
```